core/types/tx_blob.go

```diff
@@ -76,14 +76,15 @@ func (sc *BlobTxSidecar) BlobHashes() []common.Hash {
 
 // CellProofsAt returns the cell proofs for blob with index idx.
 func (sc *BlobTxSidecar) CellProofsAt(idx int) []kzg4844.Proof {
-	var cellProofs []kzg4844.Proof
-	for i := range kzg4844.CellProofsPerBlob {
-		index := idx*kzg4844.CellProofsPerBlob + i
-		if index > len(sc.Proofs) {
-			return nil
-		}
-		proof := sc.Proofs[index]
-		cellProofs = append(cellProofs, proof)
+	if sc.Version == 1 {
+		index := idx * kzg4844.CellProofsPerBlob
+		return sc.Proofs[index : index+kzg4844.CellProofsPerBlob]
```

Comment:
Ah wait this will panic if we do CellProofsAt oob

Reply:
i would prefer to return an error if out-of-range occurs, or the unexpected version

Reply:
I don't think the out-of-range can occur, because the sidecar was checked by `validateBlobSidecarOsaka` when put into the blob pool.

Reply:
Added validation for the `idx` parameter to avoid out-of-range occurs.